src/Access/SettingsProfilesInfo.cpp

```diff
@@ -55,4 +62,26 @@ SettingsProfilesInfo::getConstraintsAndProfileIDs(const std::shared_ptr<const Se
     return res;
 }
 
+Strings SettingsProfilesInfo::getProfileNames() const
+{
+    Strings result;
+    result.reserve(profiles_with_implicit.size());
+    for (const auto & profile_id : profiles)
+    {
+        const auto p = names_of_profiles.find(profile_id);
+        if (p == names_of_profiles.end())
+        {
+            if (const auto name = access_control.tryReadName(profile_id))
+                // We could've updated cache here, but it is a very rare case, so don't bother.
+                result.push_back(*name);
+            else
+                throw Exception(ErrorCodes::LOGICAL_ERROR, "Unable to get profile name for {}", toString(profile_id));
+        }
+
+        result.push_back(p->second);
```

Comment:
If `profile-id` is not found in `names_of_profiles` but in `access_control` (l. 76), then we fallthrough to l. 81 where the beyond-the-end iterator is dereferenced. The code should wrap l. 81 in an `else`. Also suggest to handle the common case first, i.e. check for `p != names_of_profiles.end()` in l. 72.